### ISSUE TITLE
About 10 clips testing

### ISSUE THREAD

**Issue by Ramsden1** [facebookresearch/SlowFast#428]
Thanks for the great codebase, I have some problems, could you help me figure out?

As described in the slowfast paper, **For testing, we uniformly sample 10 clips from a video along its temporal axis. We average the softmax scores for prediction.** , For training, I can understand the meaning of sample one clip from one video, i.e. in code, input is (batch x channel x frames x height x width). But for testing, I don't know how to represent these 10 clips for network input, is it sample one clip for 10 times? I tried to understand the code file of slowfast/datasets/kinetics.py, I found that for testing, the implementation only returned one clip from 10 clips specified by variable **temporal_sample_index**, so where is the left 9 clips? if these clips returned, could you explain how to use these clips to get video level prediction? is 10 clips of one video as a mini-batch as network input?

Thanks a lot

**Comment by haooooooqi** [comment]
Hi @Ramsden1,

Thanks for playing with PySF. If you take a further look at the temporal_sample_index, you will find we use it to *uniformly* sampled 10 clips.

Thanks,
Haoqi

**Comment by Note-Liu** [comment]
I also have the doubt，In test file，if 10 clip have different predict result，i want get the predict label of the video, which the predict result can be the final label??

**Comment by Ramsden1** [comment]
> 
> 
> I also have the doubt，In test file，if 10 clip have different predict result，i want get the predict label of the video, which the predict result can be the final label??

Hi, the default setting of slowfast is take the SUM of them to predict final video prediction. you also can take MAX by changing [this variable](https://github.com/facebookresearch/SlowFast/blob/master/slowfast/config/defaults.py#L482)

**Comment by Note-Liu** [comment]
> > I also have the doubt，In test file，if 10 clip have different predict result，i want get the predict label of the video, which the predict result can be the final label??
> 
> Hi, the default setting of slowfast is take the SUM of them to predict final video prediction. you also can take MAX by changing [this variable](https://github.com/facebookresearch/SlowFast/blob/master/slowfast/config/defaults.py#L482)


I find this code:
    top_max_k_correct = top_max_k_inds.eq(rep_max_k_labels)
    # Compute the number of topk correct predictions for each k.
    topks_correct = [top_max_k_correct[:k, :].float().sum() for k in ks]

it use SUM to get acc, but i want get the label of  one  video.what's your mean of "take the SUM of them to predict final video prediction"?

**Comment by Note-Liu** [comment]
> > I also have the doubt，In test file，if 10 clip have different predict result，i want get the predict label of the video, which the predict result can be the final label??
> 
> Hi, the default setting of slowfast is take the SUM of them to predict final video prediction. you also can take MAX by changing [this variable](https://github.com/facebookresearch/SlowFast/blob/master/slowfast/config/defaults.py#L482)

Solved ！Thanks for your reply!

**Comment by Note-Liu** [comment]
> > I also have the doubt，In test file，if 10 clip have different predict result，i want get the predict label of the video, which the predict result can be the final label??
> 
> Hi, the default setting of slowfast is take the SUM of them to predict final video prediction. you also can take MAX by changing [this variable](https://github.com/facebookresearch/SlowFast/blob/master/slowfast/config/defaults.py#L482)

I get the video final prediction,it is use SUM to get the each category result,but how do I get the video path that corresponding these final predict labels?

**Comment by Ramsden1** [comment]
> 
> 
> > > I also have the doubt，In test file，if 10 clip have different predict result，i want get the predict label of the video, which the predict result can be the final label??
> > 
> > 
> > Hi, the default setting of slowfast is take the SUM of them to predict final video prediction. you also can take MAX by changing [this variable](https://github.com/facebookresearch/SlowFast/blob/master/slowfast/config/defaults.py#L482)
> 
> I get the video final prediction,it is use SUM to get the each category result,but how do I get the video path that corresponding these final predict labels?
1
I think you should check the test set dataloder, In the __getitem__ function of kinetics400 dataset class, you can return path that dataset loaded from csv file. After that you can do for loop to iterate the testset dataloader, it will tell you what exactlly returned by test dataloader.

**Comment by Note-Liu** [comment]
> > > > I also have the doubt，In test file，if 10 clip have different predict result，i want get the predict label of the video, which the predict result can be the final label??
> > > 
> > > 
> > > Hi, the default setting of slowfast is take the SUM of them to predict final video prediction. you also can take MAX by changing [this variable](https://github.com/facebookresearch/SlowFast/blob/master/slowfast/config/defaults.py#L482)
> > 
> > 
> > I get the video final prediction,it is use SUM to get the each category result,but how do I get the video path that corresponding these final predict labels?
> > 1
> > I think you should check the test set dataloder, In the **getitem** function of kinetics400 dataset class, you can return path that dataset loaded from csv file. After that you can do for loop to iterate the testset dataloader, it will tell you what exactlly returned by test dataloader.

OK.Thanks for your reply!
